nevergrad/functions/irrigation/irrigation.py

```diff
@@ -10,77 +10,69 @@
 """
 
 
+import json
+import logging
+import os
+import urllib.request
 from pathlib import Path
-import urllib.request  # Necessary for people who will uncomment the part using data under EUPL license.
-import numpy as np
+
 import nevergrad as ng
-from ..base import ArrayExperimentFunction
-import os
+import numpy as np
 import pandas as pd
 import yaml
+from nevergrad.functions.irrigation.common_path import IRRIGATION_DATA_DIR, IRRIGATION_DIR
+from pcse.base import ParameterProvider
+from pcse.db import NASAPowerWeatherDataProvider
 from pcse.fileinput import CABOFileReader, YAMLCropDataProvider
 from pcse.models import Wofost72_WLP_FD
-from pcse.db import NASAPowerWeatherDataProvider
 from pcse.util import WOFOST72SiteDataProvider
-from pcse.base import ParameterProvider
-from nevergrad.functions.irrigation.common_path import IRRIGATION_DATA_DIR
 
+from ..base import ArrayExperimentFunction
 
 # pylint: disable=too-many-locals,too-many-statements
+with open(IRRIGATION_DIR / "known_geoloc.json") as fhandle:
+    KNOWN_GEOLOCS = json.load(fhandle)
 
 
 class Irrigation(ArrayExperimentFunction):
     variant_choice = {}
-    def __init__(self, symmetry: int) -> None:
-        data_dir = Path(__file__).with_name("data")
+
+    def __init__(self, symmetry: int, n_iterations: int = 1000) -> None:
         urllib.request.urlretrieve(
-           "https://raw.githubusercontent.com/ajwdewit/pcse_notebooks/master/data/soil/ec3.soil",
-           Path(IRRIGATION_DATA_DIR, "soil/ec3.soil"),
+            "https://raw.githubusercontent.com/ajwdewit/pcse_notebooks/master/data/soil/ec3.soil",
```

Comment:
should we make this retrieve only when the data is not already downloaded ? or only if the user specify it ?

Reply:
I would say the first option. But let's keep it simple for the moment. As the download time is quite quick we can just download at every moment the program is runned.